### PR TITLE
custom segue to present profileView from left to right

### PR DESCRIPTION
**Pull request by smanini1**
#### Description:

Created custom segue to present profileView from left to right and unwind view the same way

---

#### Preview:

ezgif com-video-to-gif (11)

### CODE REVIEW

File: ios-target/View/CustomControls/UIStoryboardSegueFromLeftToRight.swift [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+class UIStoryboardSegueFromLeftToRight: UIStoryboardSegue {
+  
+  override func perform() {
+    let sourceController = source as UIViewController
+    let destinationController = destination as UIViewController
+  
+    sourceController.view.superview?.insertSubview(destinationController.view,
+                                                   aboveSubview: sourceController.view)
+    destinationController.view.transform = CGAffineTransform(translationX: -sourceController.view.frame.size.width, y: 0)
```
Comment (mcousillas6) [inline ios-target/View/CustomControls/UIStoryboardSegueFromLeftToRight.swift]: mind indenting these?

File: ios-target/View/CustomControls/UIStoryboardSegueFromLeftToRight.swift [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    let sourceController = source as UIViewController
+    let destinationController = destination as UIViewController
+    
+    sourceController.view.superview?.insertSubview(destinationController.view,
+                                                   belowSubview: sourceController.view)
+    sourceController.view.transform = CGAffineTransform(translationX: 0, y: 0)
+    
+    UIView.animate(withDuration: 0.35,
+                   animations: {
+                    sourceController.view.transform = CGAffineTransform(translationX: -sourceController.view.frame.size.width, y: 0)
```
Comment (mcousillas6) [inline ios-target/View/CustomControls/UIStoryboardSegueFromLeftToRight.swift]: same here, try indenting consistently

File: ios-target/View/CustomControls/UIStoryboardSegueFromLeftToRight.swift [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  
+  override func perform() {
+    let sourceController = source as UIViewController
+    let destinationController = destination as UIViewController
+  
+    sourceController.view.superview?.insertSubview(destinationController.view,
+                                                   aboveSubview: sourceController.view)
+    destinationController.view.transform = CGAffineTransform(translationX: -sourceController.view.frame.size.width, y: 0)
+    
+    UIView.animate(withDuration: 0.35,
```
Comment (mcousillas6) [inline ios-target/View/CustomControls/UIStoryboardSegueFromLeftToRight.swift]: try leaving `0.35` as a constant somewhere

File: ios-target/View/CustomControls/UIStoryboardSegueFromLeftToRight.swift [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  
+  override func perform() {
+    let sourceController = source as UIViewController
+    let destinationController = destination as UIViewController
+    
+    sourceController.view.superview?.insertSubview(destinationController.view,
+                                                   belowSubview: sourceController.view)
+    sourceController.view.transform = CGAffineTransform(translationX: 0, y: 0)
+    
+    UIView.animate(withDuration: 0.35,
```
Comment (mcousillas6) [inline ios-target/View/CustomControls/UIStoryboardSegueFromLeftToRight.swift]: same

File: ios-target/View/Home/HomeViewController.swift [hunk 5]
```diff
@@ -46,6 +46,8 @@ class HomeViewController: UIViewController {
     viewController.viewModel.topics = viewModel.topics
   }
   
+  @IBAction func prepareForUnwind(segue: UIStoryboardSegue) { }
```
Comment (mato2593) [inline ios-target/View/Home/HomeViewController.swift:49]: Do you need this action?

File: ios-target/View/CustomControls/UIStoryboardSegueFromLeftToRight.swift (hunk 3 above)
Comment (mcousillas6) [inline ios-target/View/CustomControls/UIStoryboardSegueFromLeftToRight.swift]: Also, if you want, you can try playing a bit with [animationOptions](https://developer.apple.com/documentation/uikit/uiview/animationoptions) like `curveEaseInOut` to get your transition to look smoother 😁.

**Review by mcousillas6 — COMMENTED** [review]
Looking good!

**Review by mato2593 — APPROVED** [review]
Smooth!

File: ios-target/View/CustomControls/UIStoryboardSegueFromLeftToRight.swift (hunk 2 above)
Comment (mato2593) [reply ios-target/View/CustomControls/UIStoryboardSegueFromLeftToRight.swift]: I don't know why, but I think Xcode leaves just 1 space for indentation in this cases

**Review by glm4 — APPROVED** [review]

**Comment by germanStabile** [comment]
Weirdest back button evah

**Review by germanStabile — APPROVED** [review]

**Comment by mato2593** [comment]
> Weirdest back button evah

I thought the same at first, but it actually makes sense haha

**Comment by germanStabile** [comment]
> > Weirdest back button evah
> 
> I thought the same at first, but it actually makes sense haha

35cqzs

**Comment by mato2593** [comment]
> > > Weirdest back button evah
> > 
> > 
> > I thought the same at first, but it actually makes sense haha
> 
> 35cqzs

Think about how the regular back button works, it's the opposite animation, so it does make sense that the button is backwards. The arrow points to where you want to go

**Comment by germanStabile** [comment]
Then the animation is wrong... The back button will be ok if both view controllers would be side by side but in this case the profile view controller is presented on top of the map so it doesn't make sense

**Comment by mato2593** [comment]
> Then the animation is wrong... The back button will be ok if both view controllers would be side by side but in this case the profile view controller is presented on top of the map so it doesn't make sense

I agree that it would be better if both controllers were sliding to the side

**Comment by glm4** [comment]
> > Then the animation is wrong... The back button will be ok if both view controllers would be side by side but in this case the profile view controller is presented on top of the map so it doesn't make sense
> 
> I agree that it would be better if both controllers were sliding to the side

Don't fight guys. It is our minds playing tricks on us. The directions of the arrow and animations are following the same convention of the normal push/pop, but the way the views slide is not the same that's why it looks weird. I think 🤔

**Comment by germanStabile** [comment]
I demand a trial by Fifa!

File: ios-target/View/Home/HomeViewController.swift (hunk 5 above)
Comment (smanini1) [reply ios-target/View/Home/HomeViewController.swift:49]: Without this action it won't let me access the exit point on the storyboard to dismiss this view

File: ios-target/View/CustomControls/UIStoryboardSegueFromLeftToRight.swift (hunk 2 above)
Comment (smanini1) [reply ios-target/View/CustomControls/UIStoryboardSegueFromLeftToRight.swift]: yup, should i change it?

**Pull request merged**